### ISSUE TITLE
SDXL Refiner adds image watermarks almost 80% of the time with Compel

### ISSUE THREAD

**Issue by bghira** [damian0815/compel#47]
I'm not trying to overwhelm ya, I just needed this to be tracked in some way, as this experience is very strange, and off-putting.

Using the latest Compel 'exploratory version' from today, the following is what I get from the SDXL base, with Compel embeds:

![image](https://github.com/damian0815/compel/assets/59658056/7b5f6f20-418b-4c42-9719-01cf63bfff46)

There's no watermarks, or a hint of any watermarks.

Running it through laplace filter doesn't reveal any issues:

![image](https://github.com/damian0815/compel/assets/59658056/fc08d8a3-e901-4fde-93e5-3b0d9e47de9d)

Now, when running the refiner using a different Compel instance that just loads the 2nd text encoder, and retrieves pooled embeds:

![image](https://github.com/damian0815/compel/assets/59658056/ba1c0e04-e94a-4e67-84d0-8cb7d72102ef)

The laplace filter hilights the anomalies:

![image](https://github.com/damian0815/compel/assets/59658056/ee2f8372-e5bf-47c0-8a4b-34cc06b9af70)

**Comment by bghira** [comment]
I extensively tested with different sampler & scheduler combinations and every single one of them result in watermarks:

* DDIM
* UniPC
* DPM++ 2M
* DDPM
* Euler (and Ancestral variant)

Karras sigmas change nothing about this.

I am somewhat concerned that we're maybe pulling from a layer-too-early for the embeds. Diffusers throws away the final layer on import, correct? Just wondering if that was accounted for.

**Comment by damian0815** [comment]
Interesting. I don't think diffusers will be throwing away the last text layer on load with SDXL (note that this functionality worked with SD2.1 by puttying `23` in the `config.json` for the number of layers in the CLIP text encoder, rather than the expected `24`) (i do not know how many layers the SDXL encoders have)

to be clear, this is a plain text prompt with no compel syntax? or is there a `++` or something in the prompt somewhere?

**Comment by bghira** [comment]
correct, merely using the embeds has this effect. i was digging into it more last night, but i still have no answers. i have observed it without compel too, so. really unclear whats wrong

**Comment by damian0815** [comment]
i guess diffusers must be doing something internally in their prompt parsing that i'm not doing in compel. i will have to dig into the diffusers code to find out what (or i would welcome anyone else wanting to dig in on my behalf! ha)
